lib/index.d.ts

```diff
@@ -1,7 +1,40 @@
 import { Plugin } from 'webpack'
+import {
+  CompilerOptions,
+  parseComponent,
+  compile,
+  compileToFunctions,
+  ssrCompile,
+  ssrCompileToFunctions,
+  generateCodeFrame
+} from 'vue-template-compiler'
 
 declare namespace VueLoader {
   class VueLoaderPlugin extends Plugin {}
+
+  interface VueLoaderTemplateCompiler {
+    parseComponent?: typeof parseComponent
+    compile?: typeof compile
+    compileToFunctions?: typeof compileToFunctions
+    ssrCompile?: typeof ssrCompile
+    ssrCompileToFunctions?: typeof ssrCompileToFunctions
+    generateCodeFrame?: typeof generateCodeFrame
+  }
+
+  interface VueLoaderOptions {
+    transformAssetUrls?: { [tag: string]: string | Array<string> }
+    compiler?: VueLoaderTemplateCompiler
```

Comment:
`VueTemplateCompiler` type can be imported from `@vue/component-compiler-utils`.
https://github.com/vuejs/component-compiler-utils/blob/442d7b221d2075b2b53d8f6d4e1a4b4e6ab911e3/lib/types.ts#L16-L28

Reply:
Thanks! I noticed a mismatch between the compiler options from [component-compiler-utils](https://github.com/vuejs/component-compiler-utils/blob/442d7b221d2075b2b53d8f6d4e1a4b4e6ab911e3/lib/types.ts#L33) and from [vue-template-compiler](https://github.com/ktsn/vue/blob/6449085949305eefad08402cbf7905f2e67286ff/packages/vue-template-compiler/types/index.d.ts#L6). Which one you think is more appropriate for me to use?

Reply:
The latest one in the core repo is more complete: https://github.com/vuejs/vue/blob/29e745f99f43d0629df547da80b016996ae70872/packages/vue-template-compiler/types/index.d.ts#L6

Both are fine, though. As all the fields are optional.